subxt/src/rpc/rpc.rs

```diff
@@ -362,11 +362,37 @@ impl<T: Config> Rpc<T> {
                 rpc_params![function, to_hex(call_parameters), at],
             )
             .await?;
+        Ok(bytes)
+    }
+
+    /// Execute a runtime API call and decode the result.
+    pub async fn state_call<Res: Decode>(
+        &self,
+        function: &str,
+        call_parameters: Option<&[u8]>,
+        at: Option<T::Hash>,
+    ) -> Result<Res, Error> {
+        let bytes = self.state_call_raw(function, call_parameters, at).await?;
         let cursor = &mut &bytes[..];
         let res: Res = Decode::decode(cursor)?;
         Ok(res)
     }
 
+    /// Execute runtime API call and return the specified runtime metadata version.
+    pub async fn metadata_at_version(&self, version: u32) -> Result<Metadata, Error> {
+        let param = version.encode();
+        let opaque: Option<frame_metadata::OpaqueMetadata> = self
+            .state_call("Metadata_metadata_at_version", Some(&param), None)
+            .await?;
+
+        let bytes = opaque.ok_or(Error::Other("Metadata version not found".into()))?;
+
+        let meta: RuntimeMetadataPrefixed = Decode::decode(&mut &bytes.0[..])?;
+
+        let metadata: Metadata = meta.try_into()?;
+        Ok(metadata)
+    }
```

Comment:
The CLI tool has a fallback to use the older methods if the new ones aren't available.

I'm a bit worried that if we merge this, Subxt will break for everybody running on a slightly older node until the new metadata interface is available everywhere.

I think for now that we should fall back to fetching the old version if we can't fetch the new one, either here or in `fetch_metadata` (we could then have a call like this `metadata_at_version` that just fetches the old one, and then in `fetch_metadata` try one after the other, for instance. At least that would give us some wiggle room so that Subxt would continue to work with older versions of Substrate.